### PR TITLE
feat: enabled new styling panel for sn table

### PR DESCRIPTION
**Pull request by rajani-varpe**
A new styling panel has been designed to be created in react, replacing the old style editor. Therefore, the sn-table- properties has been updated to allow new react components to be used to construct the new styling panel in React. This PR is about that particular property definition.This Feature is not under Feature Flag.
Also added translations for labels i.e for Header, Content and Row Hover.

image

### CODE REVIEW

File: src/ext.js
Comment (haxxmaxx) [inline src/ext.js:110]: `createStylingEditorDefinition` seems to be the old definition, right? The names doesn't really differ to me. they are both editors and panels in a sense.

Comment (haxxmaxx) [inline src/ext.js:110]: Is the naming here that the component prop is `panel-section` for the new one and `style-editor` for the old?

File: src/styling-panel-property-definition.js
Comment (haxxmaxx) [inline src/styling-panel-property-definition.js:1]: This is static at this point so does not need to be a function, same for the old definition

File: src/ext.js
Comment (haxxmaxx) [inline src/ext.js:108]: This is failing in the unit tests since isEnabled is false. Easiest would be to add optional chaining and the tests would run with `createStylingEditorDefinition`

**Comment by LiKang6688** [comment]
@rajani-varpe To be clear, the straight table and the sn-table (supernova table) are different.

File: src/ext.js
Comment (rajani-varpe) [reply src/ext.js:110]: For the new one it is `styling-panel`.

File: src/ext.js
Comment (rajani-varpe) [reply src/ext.js]: Added optional chaining

File: src/styling-panel-property-definition.js
Comment (rajani-varpe) [reply src/styling-panel-property-definition.js]: Removed functions from both old and new definition.

File: src/ext.js
Comment (niekvanstaveren) [inline src/ext.js:108]: I don't think there is a need to have this behind a feature flag. The sn-table is not available in the client yet (or at least not the property panel for it).

File: src/styling-panel-property-definition.js
Comment (niekvanstaveren) [inline src/styling-panel-property-definition.js:1]: This is not a function anymore, so should probably have a different name

File: src/styling-panel-property-definition.js
Comment (niekvanstaveren) [inline src/styling-panel-property-definition.js:30]: Theme should be available through the environment. Have a look if you can access the styleService:
Theme.initializeService

File: src/styling-panel-property-definition.js
Comment (rajani-varpe) [reply src/styling-panel-property-definition.js:1]: Made required changes

File: src/ext.js
Comment (niekvanstaveren) [reply src/ext.js:108]: No when we add this to the client we will use the new panel directly, so we won't have to keep this behind a feature flag. We can directly clean up the old parts as well

File: src/styling-panel-property-definition.js
Comment (rajani-varpe) [reply src/styling-panel-property-definition.js:30]: We do not get the initializeService() access from env.

File: src/ext.js
Comment (rajani-varpe) [reply src/ext.js]: Removed the old styling panel def along with the feature flag

File: src/styling-panel-property-definition.js
Comment (haxxmaxx) [reply src/styling-panel-property-definition.js]: IMO this can stay commented for now, and fixed in a future PR

**Comment by haxxmaxx** [comment]
Tested quickly by building from this branch and importing it as an extension, had the flag on as well. Doesn't look right to me
Screenshot 2022-08-29 at 14 05 34

**Comment by rajani-varpe** [comment]
> Tested quickly by building from this branch and importing it as an extension, had the flag on as well. Doesn't look right to me
> Screenshot 2022-08-29 at 14 05 34

Actually the "Integer" component is implemented in Analytics repo and we had already raised a PR for it there which is not yet merged due to which you are getting the error of "No component found as Integer".

**Comment by haxxmaxx** [comment]
> Actually the "Integer" component is implemented in Analytics repo and we had already raised a PR for it there which is not yet merged due to which you are getting the error of "No component found as Integer".

Ok so should we wait for that to be merged before we merge this PR?

**Comment by rajani-varpe** [comment]
> > Actually the "Integer" component is implemented in Analytics repo and we had already raised a PR for it there which is not yet merged due to which you are getting the error of "No component found as Integer".
> 
> Ok so should we wait for that to be merged before we merge this PR?

Yes, as this table is not under any feature flag I think we should wait for the Integer component PR to get merge else it will give "Integer component not found" error.

Will inform you once the Integer component PR gets merged.

**Comment by rajani-varpe** [comment]
> > > Actually the "Integer" component is implemented in Analytics repo and we had already raised a PR for it there which is not yet merged due to which you are getting the error of "No component found as Integer".
> > 
> > 
> > Ok so should we wait for that to be merged before we merge this PR?
> 
> Yes, as this table is not under any feature flag I think we should wait for the Integer component PR to get merge else it will give "Integer component not found" error.
> 
> Will inform you once the Integer component PR gets merged.

Integer Component PR is merged, can you please check and confirm, so that I will squash and merge this PR as well.

File: src/ext.js
Comment (haxxmaxx) [inline src/ext.js:109]: This might be changed in the future, since the customer-facing name for sn-table is not decided yet. But this is fine for now

**Review by haxxmaxx — APPROVED** [review]

**Review by AmeliaAndersson — CHANGES_REQUESTED** [review]
Screenshot 2022-09-09 at 16 09 31

The default values/placeholders are incorrect. These should not be 0 and white but instead the actual font size that is being used for header and content and the colour used for the text. If qlik-oss doesnt use theme Id rather see these values be hardcoded than to be 0 and white. Maybe we can look in the sn-table code and see how they retrieve the default values.

**Comment by AmeliaAndersson** [comment]
In the old one its just empty in the integer and the colorpickerbox is set to "none", could we have that instead? I think that looks better than 0 and white.

**Comment by rajani-varpe** [comment]
> Screenshot 2022-09-09 at 16 09 31
> 
> The default values/placeholders are incorrect. These should not be 0 and white but instead the actual font size that is being used for header and content and the colour used for the text. If qlik-oss doesnt use theme Id rather see these values be hardcoded than to be 0 and white. Maybe we can look in the sn-table code and see how they retrieve the default values.

As mentioned in previous review comment in this PR, we do not have theme access in oss repo due to which we have commented the placeholder line as it is which is already confirmed by Niek.

image

File: src/ext.js
Comment (rajani-varpe) [reply src/ext.js:109]: Yes sure

**Comment by haxxmaxx** [comment]
But the old one didn't show the wrong value, and the color picker showed the "none" color. @AmeliaAndersson is the none color available in the new picker yet?
Screenshot 2022-09-12 at 08 23 07

**Comment by AmeliaAndersson** [comment]
> But the old one didn't show the wrong value, and the color picker showed the "none" color. @AmeliaAndersson is the none color available in the new picker yet? Screenshot 2022-09-12 at 08 23 07

Yes, you can have none in the new colorpicker. Its the default and then you can chose "disablenone" if you dont want it.

**Comment by rajani-varpe** [comment]
> disablenone

@AmeliaAndersson I am not able to use defaultNone in color picker in oss/sn-table definition, it does not shows any effect on color-picker look. And in the placeholder/value do hardcoded value work ?

**Comment by AmeliaAndersson** [comment]
I tried it out and seems theres a bug in the hook for disableNone in the colorpicker component in analytics folder. I will look at it and create a Jira bug for it. We can merge this in the meanwhile and Ill fix it in the future when I fix the colorpicker.

**Review by AmeliaAndersson — APPROVED** [review]

**Pull request merged**
